Depend on latest kube version

humblec opened the pull request:
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>


-->
```release-note

csi-lib-utils library package dependencies are updated to make use of latest kube version v1.21.0 and also v1.4.0 version of csi spec which include many bug fixes and enhancements, with this release. 

```

pohly commented:
Please check the test failures.

humblec commented:
> Please check the test failures.

Sure.. checking on it @pohly , interestingly before sending the PR I checked make and it worked. The PR fails on` make test` , and it seems to be caused by the CSI spec upgrade. Working on it :)

humblec commented:
@pohly it should be good to go, ptal  :+1: .. Thanks !

pohly commented:
Please add the changes from your last commit to the commits where they belong (CSI update for the new method, not sure about the go.sum change). Or squash everything - your choice.

pohly commented:
Please also add a release note entry.

/kind cleanup

humblec commented:
> Please also add a release note entry.
> 
> /kind cleanup

Squashed commits and also added release note. Please suggest if there any changes required for release note.

pohly commented on go.mod:
```diff
@@ -3,61 +3,24 @@ module github.com/kubernetes-csi/csi-lib-utils
-replace k8s.io/metrics => k8s.io/metrics v0.19.0
+replace k8s.io/component-base => k8s.io/component-base v0.21.1
 
 replace k8s.io/node-api => k8s.io/node-api v0.19.0
```
Why is node-api still v0.19.0?

humblec replied:
@pohly upgraded to 0.21.1 . Thanks! @pohly ptal.

pohly reviewed (approved):
/lgtm

pohly commented:
/approve
